### ISSUE TITLE
Multiline po file (question)

### ISSUE THREAD

**Issue by adammbalogh** [raulferras/PHP-po-parser#41]
I have a ```po``` file:
```
msgid ""
"%user% acaba de responder tu comentario.<br>Consulta que te ha "
"dicho %link%aquí</a>."
msgstr ""
"%user% acaba de respondre el teu comentari.<br>Consulta que t'ha "
"dit %link%aquí</a>."
```
This is the result:
```php
array(1) {
  '%user% acaba de responder tu comentario.<br>Consulta que te ha dicho %link%aquí</a>.' =>
  array(2) {
    'msgid' =>
    array(3) {
      [0] =>
      string(0) ""
      [1] =>
      string(63) "%user% acaba de responder tu comentario.<br>Consulta que te ha "
      [2] =>
      string(22) "dicho %link%aquí</a>."
    }
    'msgstr' =>
    array(3) {
      [0] =>
      string(0) ""
      [1] =>
      string(65) "%user% acaba de respondre el teu comentari.<br>Consulta que t'ha "
      [2] =>
      string(20) "dit %link%aquí</a>."
    }
  }
}
```

Only the array key is a concatenated ```msgid```. Is it intented?

Would not be better to concatenate all the things? Is there a case when we might need this "raw" form?

**Comment by raulferras** [comment]
The array key is just there to be able to locate translations.

Say you want to get a given translation, and you know its key is "Hello user".
You can quickly get the translation with

    $entries = $poparser->parse();
    $myKey = 'Hello user';
    if (isset($entries[$myKey])) {
        $entries[$mykey]['msgstr'] = 'Hola usuario';
    }

In your case, where the msgid is multilined, you should first "join" the msgid to get the key that will let you reach the desired translation:

    $msgid = array('Hello', 'user');
    $msgid_key = implode('', $mgid);
   if (isset($entries[$msgid_key])) {
        $entries[$msgid_key]['msgstr'] = 'Hola usuario';
    }

**Comment by adammbalogh** [comment]
Okay, I see, thank you.

**Comment by adammbalogh** [comment]
```
msgid "%s post restored from the Trash."
msgid_plural ""
"%s posts restored "
"from the Trash."
msgstr[0] ""
"%s entrada restaurada "
"desde la Papelera."
msgstr[1] "%s entradas restauradas desde la Papelera."
```

```php
array(1) {
  '%s post restored from the Trash.' =>
  array(4) {
    'msgid' =>
    array(1) {
      [0] =>
      string(32) "%s post restored from the Trash."
    }
    'msgid_plural' =>
    array(3) {
      [0] =>
      string(0) ""
      [1] =>
      string(18) "%s posts restored "
      [2] =>
      string(15) "from the Trash."
    }
    'msgstr' =>
    array(4) {
      [0] =>
      string(0) ""
      [1] =>
      string(22) "%s entrada restaurada "
      [2] =>
      string(18) "desde la Papelera."
      [3] =>
      string(42) "%s entradas restauradas desde la Papelera."
    }
  }
}
```

Okay, now I have the proper example for my original problem.

I would like to concatenate the multiline translations in my ```msgstr```array. You can see 4 records there. How could I do that when I don't know where ends the first multiline translation?

**Comment by RaulViscaweb** [comment]
I don't understand your question. If you want to concatenate the 'msgstr' content just do 

    implode('', $entries[$key]['msgstr']);

**Comment by raulferras** [comment]
I don't fully understand your question, but if all you want to do is concatenate a set of data in an entry just use the php function [implode](http://php.net/implode):

    $msgstr = implode('', $entries[$yourMsgID]['msgstr'];

**Comment by adammbalogh** [comment]
```
'msgstr' =>
    array(4) {
      [0] =>
      string(0) ""
      [1] =>
      string(22) "%s entrada restaurada "
      [2] =>
      string(18) "desde la Papelera."
      [3] =>
      string(42) "%s entradas restauradas desde la Papelera."
    }
```

In the example above there are two translations: 0,1,2 and 3

I'd like to do concatenating individually on each translation line and I need two separate strings.

**Comment by raulferras** [comment]
Ok now I see, there's plurals in there...
That's a bug then because the parser is mixing together every plural translation

**Comment by raulferras** [comment]
Could you provide me a po file example?

**Comment by adammbalogh** [comment]
```
msgid ""
"%s post restored from "
"the Trash."
msgid_plural ""
"%s posts restored "
"from the Trash 2."
msgstr[0] ""
"%s entrada restaurada "
"desde la Papelera."
msgstr[1] "%s entradas restauradas desde la Papelera."
```

**Comment by raulferras** [comment]
Fixed on master.
Will port the fix to 5.0 branch too.
